### DISCUSSION TITLE
How to make this type of canvas rotation possible?

### DISCUSSION THREAD

**Discussion by edthoms** [fabricjs/fabric.js#8314] [Q&A]
Hello, 
 I want to make a canvas that is rotatable, zoomable and panable; just like those in Procreate or Photoshop. [Like this video example](https://www.dropbox.com/s/27hbcgmychi5kwj/canvasManipulation.mov?dl=0)
 
What I was thinking is if I could make a white rectangle inside a canvas with grey background, that somehow is also the only thing that catches the drawing & erasing commands, as in, drawing outside the rectangle is ignored, but drawing inside of it is accepted. 
 
 And also, the positioning and size of every path inside the rectangle should be maintained (and scaled up & down accordingly)  when the rectangle is manipulated (rotated, zoomed, panned). 
 
 Is there any way of making this happen?
 
 (I've tried to rotate the fabric canvas itself using `interact.js`, but sadly whenever I do that, the drawing system gets all confused and what was already drawn goes haywire, and also whatever new path is drawn when the canvas is rotated doesn't get drawn in the position the pen landed on in the visibly rotated canvas. 
 
 It seems like the visible canvas "paper" gets rotated, but the internal "input catcher" stays in the same position and scale, which is why the new pen strokes go to the place they would originally go if the canvas was in its original position and not rotated.  [Video example](https://www.dropbox.com/s/iepc5c9ncjn27al/canvasCrazy.mov?dl=0))
 
Thanks for your time.

**Comment by ShaMan123** [comment]
#4624 is a long time pending PR
#7860 and in general group #7670 are viable workaround but yet to be merged to master

If you want to pick up the canvas PR we will give a hand

**Reply by ShaMan123** [reply]
It'll better opening a new PR from scratch

**Reply by ShaMan123** [reply]
and you don't need to wait for Layer.
It is a subclass of Group. You can handle most of that yourself if you don't want to work on a release candidate
